### ISSUE TITLE
[macOS] getEnvironmentVariable:withEncoding: fails when an installation path contains wide string

### ISSUE THREAD

**Issue by tomooda** [pharo-project/pharo-launcher#269]
To reproduce, 
1. create a folder with a wide string name like "ランチャー"  and 
2. place PharoLauncher.app into it.
3. When you launch the app, 
4. an exception "Error: Can't find the requested origin" is signaled.
You'll see the FFI call to libc to get the 'HOME' env var has failed.

For a quick fix, modifying UnixEnvironment>>getEnvironmentVariable:withEncoding: into

```
getEnvironmentVariable: aVariableNameString withEncoding: anEncoding
   ^ (self at: aVariableNameString) asByteArray decodeWith: anEncoding
```

looks working ok.
Maybe we rather better fix UnixEnvironment>>at:ifAbsent: and alikes, but it may give bigger impact on other code.

**Comment by guillep** [comment]
I've created

```
Name: PharoLauncher-Env-GuillermoPolito.5
Author: GuillermoPolito
Time: 16 October 2018, 11:59:56.185732 am
UUID: cf72515a-b734-0d00-bc35-a6a902e7d1db
Ancestors: PharoLauncher-Env-GuillermoPolito.4

Skip reading sources file from HOME in osx.
This cannot be read using uFFI at startup because the sources file was not read yet. This means that uFFI will use the decompiled version of the method, and moreover, the AST generated will be cached in the subsequents executions.

This change avoids caching wrong ASTs.
```

**Comment by guillep** [comment]
We could use a solution using the VM's primitive. However, window's primitive doe snot handle multibyte characters. We need a way to make it work now without changing the VM...

**Comment by tomooda** [comment]
I confirmed that https://github.com/pharo-project/pharo-launcher/issues/269#issuecomment-430179749 works fine in my env. On Windows 10 and Ubuntu 18.04, 1.4.5 works fine.

**Comment by demarey** [comment]
Thanks for the report and the fix. It is now included in the latest version: 1.4.6.
